src/Sentry/Internal/ThreadSafeRandom.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Sentry.Internal
+{
+    internal static class ThreadSafeRandom
+    {
+        private static readonly object Lock = new();
+        private static readonly Lazy<Random> RandomLazy = new();
+
+        private static Random Random => RandomLazy.Value;
+
+        public static double NextDouble()
+        {
+            lock (Lock)
```

Comment:
`Random` is a private field here, why not use that for locking?

Reply:
It's an evaluated property here but I guess you can lock on the lazy instead. I always prefer a dedicate lock object for clarity of intent though.

Reply:
It's a good rule of thumb but this class is so simple, it's clear what we're doing here. Not locking on a public instance nor there are many instances, bug prone thing going on.

Reply:
Are we saving up anything with the Lazy here? Maybe we just `new` up straight away? it's anyway a heap allocated object, or is the random ctor expensive?

Reply:
> Are we saving up anything with the Lazy here?

Nope, I just adapted the existing code which used lazy.